### DISCUSSION TITLE
Add operator for easier try-catch

### DISCUSSION THREAD

**Discussion by AkbarDizaji** [dotnet/csharplang#8451] [Language Ideas]
# Proposal: Introduce the `?=` Operator for Simplified Exception Handling

## Summary

Introduce a new operator `?=` that allows assigning the result of a function to a tuple containing a response and an error. This operator simplifies error handling by capturing the function’s result or any exception thrown, assigning the response or error accordingly.

## Motivation

Error handling in C# often requires verbose try-catch blocks, leading to repetitive and cluttered code. The `?=` operator aims to streamline this process by providing a concise way to handle operations that may fail, capturing the result or the exception in a structured manner.

### Syntax

```csharp
(response, error) ?= SomeFunction();

**Comment by huoyaoyuan** [comment]
The whole point of exception handling is to *not catch* exceptions when you are not handling it.
When handling, any `if` block testing for result or error will immediately exceed the verbosity comparing with try-catch, and loses null safety.

**Comment by colejohnson66** [comment]
If you are doing enough try/catch blocks that it becomes a problem, your architecture might be wrong. Try/catch, as @huoyaoyuan implied, are for *exceptional* situations. If a method needs to open a file, write to it, then exit, it shouldn't do anything if the open fails because `return err` (like Go) is no different than rethrowing (`throw;`). The caller should decide what to do when things go wrong.
